scipy/stats/stats.py

```diff
@@ -3840,6 +3850,12 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate'):
         # Handle empty arrays or single observations.
         return SpearmanrResult(np.nan, np.nan)
 
+    if (axisout == 0):
```

Comment:
also what if axisout is not zero?

Reply:
I've removed the `()` too much staring at C code I guess. 
For the `axisout == 0` I was thinking that the default for `axis` will result in `axisout==0`
and from the warning above it looks like only 2-d numpy arrays are supported so the only other option is `axis=1` which would give `axisout==1`. 

AFAICS the only difference would be how to slice the array to check for constant input. I'll see if I can add the other axis to the check too.

Reply:
Ok, that looks good. I was not aware at first that axis can only be None, 0, 1 and if someone uses axis = 2, one gets an index error. It might be worth to add a better error message (it can be done in a future PR)

Reply:
It wasn't that much to add an index error subclass so I added w/the latest push. I forgot to add to the docstring in the `__init__.py` file for stats which is (one) cause of the build fail in travis. The others (build fails) are unrelated.

Reply:
Can you raise a ValueError?

Reply:
sure, that makes more sense here given the source of the error. In this case I ***don't*** think we need a specialized sub-class of `ValueError` so I'll remove that as well and update the unit test accordingly.